morpho5/builtin/veneer.c

```diff
@@ -1082,6 +1125,70 @@ value List_ismember(vm *v, int nargs, value *args) {
     return MORPHO_NIL;
 }
 
+/** Generate sets/tuples and return as a list of lists */
+value list_generatetuples(vm *v, objectlist *list, unsigned int n, tuplemode mode) {
+    unsigned int nval=list->val.count;
+    unsigned int work[2*n];
+    value tuple[n];
+    morpho_tuplesinit(list->val.count, n, work, mode);
+    objectlist *new = object_newlist(0, NULL);
+    if (!new) goto list_generatetuples_cleanup;
+    
+    while (morpho_tuples(nval, list->val.data, n, work, mode, tuple)) {
+        objectlist *el = object_newlist(n, tuple);
+        if (el) {
+            list_append(new, MORPHO_OBJECT(el));
+        } else {
+            goto list_generatetuples_cleanup;
+        }
+    }
+    
+    list_append(new, MORPHO_OBJECT(new));
```

Comment:
It's essential before calling morpho_bindobjects that ALL objects created within the C function are bound at once. Otherwise, the garbage collector may be triggered and the object deleted. We sneak on the object (and remove it in line 1148) so that the list is also bound.